pyomo/contrib/gdpopt/GDPopt.py

```diff
@@ -42,17 +47,114 @@
+        description="Tolerance for bound convergence."
+    ))
+    CONFIG.declare("iterlim", ConfigValue(
+        default=30, domain=NonNegativeInt,
+        description="Iteration limit."
+    ))
+    CONFIG.declare("strategy", ConfigValue(
+        default="LOA", domain=In(["LOA"]),
+        description="Decomposition strategy to use."
+    ))
+    CONFIG.declare("init_strategy", ConfigValue(
+        default="set_covering", domain=In([
+            "set_covering", "max_binary", "fixed_binary", "custom_disjuncts"]),
+        description="Initialization strategy to use.",
+        doc="""Selects the initialization strategy to use when generating
+        the initial cuts to construct the master problem."""
+    ))
+    CONFIG.declare("custom_init_disjuncts", ConfigList(
+        # domain=ComponentSets of Disjuncts,
+        default=None,
+        description="List of disjunct sets to use for initialization."
+    ))
+    CONFIG.declare("max_slack", ConfigValue(
+        default=1000, domain=NonNegativeFloat,
+        description="Upper bound on slack variables for OA"
+    ))
+    CONFIG.declare("OA_penalty_factor", ConfigValue(
+        default=1000, domain=NonNegativeFloat,
+        description="Penalty multiplication term for slack variables on the "
+        "objective value."
+    ))
+    CONFIG.declare("mip", ConfigValue(
+        default="gurobi",
+        description="Mixed integer linear solver to use."
+    ))
+    mip_options = CONFIG.declare("mip_options", ConfigBlock(implicit=True))
+    CONFIG.declare("nlp", ConfigValue(
+        default="ipopt",
+        description="Nonlinear solver to use"))
+    nlp_options = CONFIG.declare("nlp_options", ConfigBlock(implicit=True))
+    CONFIG.declare("master_postsolve", ConfigValue(
+        default=_DoNothing,
+        description="callback hook after a solution of the master problem"
+    ))
+    CONFIG.declare("subprob_presolve", ConfigValue(
+        default=_DoNothing,
+        description="callback hook before calling the subproblem solver"
+    ))
+    CONFIG.declare("subprob_postsolve", ConfigValue(
+        default=_DoNothing,
+        description="callback hook after a solution of the "
+        "nonlinear subproblem"
+    ))
+    CONFIG.declare("subprob_postfeas", ConfigValue(
+        default=_DoNothing,
+        description="callback hook after feasible solution of "
+        "the nonlinear subproblem"
+    ))
+    CONFIG.declare("algorithm_stall_after", ConfigValue(
+        default=2,
+        description="number of non-improving master iterations after which "
+        "the algorithm will stall and exit."
+    ))
+    CONFIG.declare("tee", ConfigValue(
+        default=False,
+        description="Stream output to terminal.",
+        domain=In([True, False])
+    ))
+    CONFIG.declare("logger", ConfigValue(
+        default=logging.getLogger('pyomo.contrib.gdpopt'),
+        description="The logger object to use for reporting."
```

Comment:
What do you think of adding a domain here that will accept either a logging object or a name?  I guess I think it is a bit weird passing a logger object as an argument -- in part because I am not quite sure how to serialize it to a "configuration file"